src/bif/erlang.rs

```diff
@@ -0,0 +1,519 @@
+        };
+        if t.len != 2 {
+            return Err(Exception::new(Reason::EXC_BADARG))
+        }
+        let n = match t[0].into_number() {
+            Ok(value::Num::Integer(i)) if ! i < 1 && i - 1 < num => i - 1,
+            _ => return Err(Exception::new(Reason::EXC_BADARG))
+        };
+        unsafe {
+            std::ptr::write(&mut tuple[n as usize], t[1]);
+        }
+    }
+    Ok(Term::from(tuple))
+}
+
+pub fn bif_erlang_append_element_2(_vm: &vm::Machine, process: &RcProcess, args: &[Term]) -> BifResult {
+    if !args[0].is_tuple() {
+        return Err(Exception::new(Reason::EXC_BADARG));
+    }
+    let t: &Tuple = match args[0].try_into() {
+        Ok(tuple) => tuple,
+        _ => return Err(Exception::new(Reason::EXC_BADARG))
+    };
+    let heap = &process.context_mut().heap;
+    let mut new_tuple = value::tuple(&heap, (t.len() + 1) as u32);
+    unsafe {
+        new_tuple[..t.len()].copy_from_slice(&t[..]);
+        std::ptr::write(&mut new_tuple[t.len()], args[1]);
+    }
+    Ok(Term::from(new_tuple))
+}
+
+pub fn bif_erlang_setelement_3(_vm: &vm::Machine, process: &RcProcess, args: &[Term]) -> BifResult {
+    let number = match args[0].into_number() {
+        Ok(value::Num::Integer(i)) if ! i < 1 => i - 1,
+        _ => return Err(Exception::new(Reason::EXC_BADARG))
+    };
+    let t: &Tuple = match args[1].try_into() {
+        Ok(tuple) => tuple,
+        _ => return Err(Exception::new(Reason::EXC_BADARG))
+    };
+    if !(number <= t.len() as i32) {
+        return Err(Exception::new(Reason::EXC_BADARG));
+    }
+    let heap = &process.context_mut().heap;
+    let mut new_tuple = value::tuple(&heap, t.len() as u32);
+    unsafe {
+        new_tuple[..t.len()].copy_from_slice(&t[..]);
+        std::ptr::write(&mut new_tuple[number as usize], args[2]);
+    }
+    Ok(Term::from(new_tuple))
+}
+
+pub fn bif_erlang_tuple_to_list_1(_vm: &vm::Machine, process: &RcProcess, args: &[Term]) -> BifResult {
+    if !args[0].is_tuple() {
+        return Err(Exception::new(Reason::EXC_BADARG));
+    }
+    let t: &Tuple = match args[0].try_into() {
+        Ok(tuple) => tuple,
+        _ => return Err(Exception::new(Reason::EXC_BADARG))
+    };
+    let mut cons = vec![];
+    for &item in t.iter() {
+        cons.push(item);
+    }
+    let heap = &process.context_mut().heap;
+    Ok(iter_to_list!(&heap, cons.iter().map(|x| *x)))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::process::{self};
+    use crate::module;
+    use crate::atom;
+
+    #[test]
+    fn test_bif_erlang_make_tuple_2() {
+        let vm = vm::Machine::new();
+        let module: *const module::Module = std::ptr::null();
+        let process = process::allocate(&vm.state, module).unwrap();
+
+        let number = Term::int(2);
+        let default = str_to_atom!("test");
+        let args = vec![number, default];
+
+        let res = bif_erlang_make_tuple_2(&vm, &process, &args);
+        if let Ok(x) = res {
```

Comment:
I think we could avoid `if let` and just use `let x = res.unwrap()`, since `unwrap` will panic

Reply:
So, it only panics when the res is '''Err(x)'''?

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap

It unwraps the Ok, or panics on Err, it's not recommended in code, but it's okay to use in tests!